File: core/core-api/src/main/java/io/say/better/domain/solution/application/converter/SolutionConverter.java
package io.say.better.domain.solution.application.converter;

import io.say.better.storage.mysql.domain.entity.Member;
import io.say.better.storage.mysql.domain.entity.Solution;
import io.say.better.domain.solution.ui.dto.SolutionRequest;

public class SolutionConverter {

	private SolutionConverter() {
		throw new IllegalStateException("Utility class");
	}

	public static Solution toSolution(SolutionRequest.CreateSolution request, Member member) {
		return Solution.builder()
				.writer(member)
				.title(request.getTitle())
				.baselineSessionTime(request.getBaselineSessionTime())
				.interCompleteThresh(request.getInterCompleteThresh())
				.interMaintainTerm(request.getInterMaintainTerm())
				.commOptTimes(request.getCommOptTimes())
				.commOptCnt(request.getCommOptCnt())
				.build();
	}

}

File: core/core-api/src/main/kotlin/io/say/better/domain/solution/application/converter/SolutionConverter.kt
package io.say.better.domain.solution.application.converter

import io.say.better.domain.solution.ui.dto.SolutionRequest.CreateSolution
import io.say.better.storage.mysql.domains.account.entity.Educator
import io.say.better.storage.mysql.domains.account.entity.Learner
import io.say.better.storage.mysql.domains.solution.entity.Solution

class SolutionConverter private constructor() {
    init {
        throw IllegalStateException("Utility class")
    }

    companion object {
        fun toSolution(
            request: CreateSolution,
            educator: Educator,
            learner: Learner,
        ): Solution {
            return Solution(
                nowState = request.nowState,
                educationGoal = request.educationGoal,
                description = request.description,
                writer = educator,
                learner = learner,
                title = request.title,
                commOptTimes = request.commOptTimes,
                commOptCnt = request.commOptCnt,
            )
        }
    }
}
